packages/labs/ssr-react/src/jsx-dev-runtime.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @fileoverview To serve as JSX import source for runtime JSX transforms in
+ * development mode. For use in browsers.
+ */
+
+import 'lit/experimental-hydrate-support.js';
+
+// eslint-disable-next-line import/extensions
+export {jsxDEV} from 'react/jsx-dev-runtime';
```

Comment:
Why doesn't this have Fragment and jsxs? Should it be an `export *`?

Is there a link to docs on the module exports contract for this module?

Reply:
Good catch for the `Fragment`. Added. I think it's better to be explicit rather than `export *` so we know it exports everything that's needed. It doesn't have a separate `jsxs` because in dev mode the `jsxDEV` serves both purpose and gets a `isStaticChildren` boolean as a fourth argument instead instead.

There was no canonical doc that specifies this contract. Things I referenced were:
- React's RFC: https://github.com/reactjs/rfcs/blob/createlement-rfc/text/0000-create-element-changes.md
- Babel's release note: https://babeljs.io/blog/2020/03/16/7.9.0#a-new-jsx-transform-11154httpsgithubcombabelbabelpull11154
- React's own export: https://github.com/facebook/react/blob/9d111ffdfbcfee4b348a3d49c16f02cb718c896f/packages/react/jsx-dev-runtime.js
- Emotion's implementation: https://github.com/emotion-js/emotion/blob/acb72a45592881d9d1f72003b6db2e488b981599/packages/react/src/jsx-dev-runtime.js
- theme-ui's implementation: https://github.com/system-ui/theme-ui/blob/2d8bfc3ab29e54e925095ba0b5dbdb07bf7209db/packages/core/src/jsx-dev-runtime.ts

Reply:
I ask about `export *` because if this module needs to implement a certain contract that we know that `react/jsx-dev-runtime` does, then the easiest way to guarantee that we meet the contract is to `export *` that module. It fits the spirit of monkey-patching that module, which is kind of what we're emulating.